### ISSUE TITLE
default.xsl declares it is a 2.0 stylesheet, but it appears to have issues with a 2.0 processor

### ISSUE THREAD

**Issue by jpschewe** [spotbugs/spotbugs#958]
According to discussion on at https://github.com/spotbugs/spotbugs-gradle-plugin/issues/114 there appears to be a problem with default.xsl in the spotbugs distribution. When I try and execute default.xsl with the saxon processor I get errors
```
Warning: at xsl:variable on line 348 column 56 of default.xsl:
  SXWN9001: A variable with no following sibling instructions has no effect
Warning: at xsl:variable on line 351 column 59 of default.xsl:                                 
  SXWN9001: A variable with no following sibling instructions has no effect
Error on line 71 of default.xsl:
  SEPM0009: Values of 'standalone' and 'omit-xml-declaration' conflict   
```

**Comment by mustafashabib** [comment]
I too am not seeing how to generate html output using v2.0.0 of this plugin

**Comment by mustafashabib** [comment]
I ended up doing this in my Gradle build script:

```kotlin
tasks.withType(SpotBugsTask::class) {
        reports {
            xml.isEnabled = false
            xml.isWithMessages = true
            html.isEnabled = true
        }
        extraArgs("-html")

        tasks.register("${name}HtmlReport") {
            var input = reports.xml.destination
            val output = file(input.absolutePath.replaceFirst(".xml", ".html"))

            doLast {

                logger.debug("converting spotbugs report $input to $output")
                if (file(input).exists()) {
                    println("Processing $input to $output")
                    val factory = TransformerFactory.newInstance("net.sf.saxon.TransformerFactoryImpl", Thread.currentThread().contextClassLoader)
                    val transformer = factory.newTransformer(StreamSource(file("config/spotbugs/color.xsl")))
                    transformer.transform(StreamSource(input), StreamResult(output))
                }
            }
        }
        finalizedBy("${name}HtmlReport")
    }
```

**Comment by chrismiceli** [comment]
Created PR https://github.com/spotbugs/spotbugs/pull/1000.

**Comment by KengoTODA** [comment]
`4.0.0-beta4` contains fix introduced by #1000.

**Comment by jpschewe** [comment]
I moved to using 4.0.0-beta4 for my project using the configuration below:
```
spotbugs {
    toolVersion = "4.0.0-beta4"
}

configurations {
    spotbugsStylesheets { transitive = false }
}

dependencies {
  spotbugsStylesheets("com.github.spotbugs:spotbugs:$spotbugs.toolVersion")
}

tasks.withType(SpotBugsTask) {
    reports {
        xml.withMessages = true
        html.stylesheet = resources.text.fromArchiveEntry(configurations.spotbugsStylesheets, "default.xsl")
    }

    task "${it.name}HtmlReport" {
        def input = reports.xml.destination
        inputs.file(reports.html.stylesheet.asFile()).withPropertyName("spotbugsStylesheet").withPathSensitivity(PathSensitivity.NONE)
        inputs.files(fileTree(input)).withPropertyName("input").withPathSensitivity(PathSensitivity.NONE) skipWhenEmpty()
        def output = file(input.absolutePath.replaceFirst(/\.xml$/, ".html"))
        outputs.file(output).withPropertyName("output")
        doLast {
            def factory = TransformerFactory.newInstance("net.sf.saxon.TransformerFactoryImpl", getClass().classLoader)
            def transformer = factory.newTransformer(new StreamSource(reports.html.stylesheet.asFile()));
            transformer.transform(new StreamSource(input), new StreamResult(output))
        }
    }
    it.finalizedBy("${it.name}HtmlReport")
}
```

Using "color.xsl" works, however using "default.xsl" I get
```
Warning at xsl:variable on line 348 column 56 of default.xsl:
  SXWN9001: A variable with no following sibling instructions has no effect
Warning at xsl:variable on line 351 column 59 of default.xsl:
  SXWN9001: A variable with no following sibling instructions has no effect
Error evaluating ((<head {(<title {text{"SpotBugs Report"}}/>, ...)}/>, ...)) on line 71 column 8 of default.xsl:
  SEPM0009: Values of 'standalone' and 'omit-xml-declaration' conflict
  In template rule with match="/" on line 69 of default.xsl
```

I'm using the spotbugs gradle plugin version 2.0.1. 
What am I missing here?

**Comment by fcurts** [comment]
I'm getting this warning when running SpotBugs 4.0.0 in a Gradle 6.2.1 build with "gradle.plugin.com.github.spotbugs.snom:spotbugs-gradle-plugin-v2:0.5.1".
